crates/orchestrator/src/data_storage/aws_s3/mod.rs

```diff
@@ -0,0 +1,122 @@
+use crate::data_storage::aws_s3::config::AWSS3Config;
+use crate::data_storage::types::StarknetOsOutput;
+use crate::data_storage::DataStorage;
+use async_trait::async_trait;
+use aws_sdk_s3::config::{Builder, Credentials, Region};
+use aws_sdk_s3::primitives::ByteStream;
+use aws_sdk_s3::{Client, Error};
+
+pub mod config;
+
+pub struct AWSS3 {
+    client: Client,
+    config: AWSS3Config,
+}
+
+impl AWSS3 {
+    #[allow(dead_code)]
+    pub async fn new(config: AWSS3Config) -> Self {
+        // AWS cred building
+        let credentials = Credentials::new(
+            config.s3_key_id.clone(),
+            config.s3_key_secret.clone(),
+            None,
+            None,
+            "loaded_from_custom_env",
+        );
+        let region = Region::new(config.s3_bucket_region.clone().to_string());
+        let conf_builder = Builder::new()
+            .region(region)
+            .credentials_provider(credentials);
+        let conf = conf_builder.build();
+
+        // Building AWS S3 config
+        let client = Client::from_conf(conf);
+
+        Self { client, config }
+    }
+}
+
+#[async_trait]
+impl DataStorage for AWSS3 {
+    async fn get_data_for_block(&self, key: &str) -> Result<StarknetOsOutput, Error> {
```

Comment:
it should be just `get_data` and it should return just bytes. if we hardcode `StarknetOsOutput`, I can't use S3 later on with the DA layer for example in case I want to store it in s3